baseband/gsb/base.py

```diff
@@ -142,12 +141,25 @@ def __init__(self, fh_ts, fh_raw, header0, thread_ids=None,
             sample_shape = GSBPayload._sample_shape_maker_nthread(len(fh_raw),
                                                                   nchan)
 
+        if sample_rate is None:
+            try:
+                header1 = self.fh_ts.read_timestamp()
+                assert (fh_ts.tell() ==
+                        header0.seek_offset(2))
+                sample_rate = samples_per_frame * (
```

Comment:
I wonder if we should use knowledge of the GSB here: the "master" clock is at 200 MHz, and sample rates are integer divisions of this... (fine for another PR)